worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java

```diff
@@ -102,10 +123,21 @@ protected WorkerResponse createWorkerResponseImpl()
         documentWorkerResult.document = documentTask.document;
         documentWorkerResult.changeLog = changeLog;
 
+        ResponseOptions responseOptions = getResponseOptions();
         // Select the output queue
-        final String outputQueue = ChangeLogFunctions.hasFailures(changes)
-            ? application.getFailureQueue()
-            : application.getSuccessQueue();
+        // TODO: ResponseOptions queue name will override the queue set below. This means that failure queue will
+        // not be used in case of failures. This is correct behaviour but we might want to add ability
+        // to set a failure queue on response options.
+        String outputQueue = ChangeLogFunctions.hasFailures(changes) ? application.getFailureQueue()
+                : application.getSuccessQueue();
+
+        if (responseOptions.getQueueName() != null) {
```

Comment:
- The JavaDoc for `getQueueName()` reads "Gets the queue name that will be used when a worker sends response message."  I had thought from the description that it would return the success or failure queue if the queue wasn't explicitly set, but I see now from how it is being used that it only returns if the queue has been explicitly set.  I'm not sure I like the `getQueueName()` method name anymore.  Perhaps it should be `getQueueNameOverride()` or something like that?  Perhaps that is why you are so keen that the class is called `ResponseOptions` rather than the more definitive `Response`?
- Since the explicitly set queue name takes priority why not check it first, avoiding the need to call `ChangeLogFunctions.hasFailures()` if it is not necessary.

Reply:
I understand your point but it's not really confusing. Let's not change it... Sorry. I can't get McA to change his code over again.